test/e2e/tests/test_adopt_endpoint.py

```diff
@@ -134,6 +134,9 @@ def adopted_endpoint(sdk_endpoint):
     adopt_model_reference, _, adopt_model_resource = create_adopted_resource(
         replacements=replacements,
     )
+    adopt_model_resource = k8s.wait_resource_consumed_by_controller(
+        adopt_model_reference
+    )
```

Comment:
you're already calling `k8s.wait_resource_consumed_by_controller` in the `test_smoke` below for each of these resources...